### ISSUE TITLE
Deprecate limitedHash functionality with next major version of Fsharp.Core

### ISSUE THREAD

**Issue by T-Gro** [fsharp/fslang-suggestions#1230]
I propose we **deprecate** all functionality around **limited structural hashing** and it's corresponding comparer and hasher.

Searching for usages of it on GitHub or Google do not yield any productive results, please correct me if this is not the case.

## Pros and Cons 

The **advantages** of making this adjustment to F# are sending a strong signal about this feature not being recommended.
There aren't really tests around it in the compiler repo and it is very likely it can have some corner cases not yet observed.

The **disadvantages** of making this adjustment to F# are that people might be using it - please do let us know if that's you.

## Extra information

Estimated cost: XS

## Affidavit (please submit!)

Please tick this by placing a cross in the box:
* [ X] This is not a question (e.g. like one you might ask on [stackoverflow](http://stackoverflow.com)) and I have searched stackoverflow for discussions of this issue
* [X ] I have [searched both open and closed suggestions on this site](http://github.com/fsharp/fslang-suggestions/issues) and believe this is not a duplicate
* [X ] This is not something which has obviously "already been decided" in previous versions of F#.  If you're questioning a fundamental design decision that has obviously already been taken (e.g. "Make F# untyped") then please don't submit it.

Please tick all that apply:
* [ ] This is not a breaking change to the F# language design
* [ X] I or my company would be willing to help implement and/or test this

## For Readers

If you would like to see this issue implemented, please click the :+1: emoji on this issue. These counts are used to generally order the suggestions by engagement.

**Comment by cartermp** [comment]
Or it could be renamed to possiblyGiveUpHash 🙂

**Comment by abelbraaksma** [comment]
Just to be sure, we're talking about this obscure function, right? https://github.com/dotnet/fsharp/blob/bc88b51bec7cd3a76d0e4b2eb2f4b5aa1e8b50ca/src/FSharp.Core/prim-types.fs#L5172

> Searching for usages of it on GitHub or Google do not yield any productive results, please correct me if this is not the case.

It does exist in the [frozen docs](https://github.com/MicrosoftDocs/visualfsharpdocs/blob/main/docs-fsharp-conceptual/operators.limitedhash%5B't%5D-function-%5Bfsharp%5D.md).

On GitHub it seems to only appear in (old) forks. Recently, [DeviousAsti tweeted about it](https://twitter.com/deviousasti/status/1588762449259683842?ref_src=twsrc%5Etfw), but yeah, nothing substantial.

**Comment by abelbraaksma** [comment]
Ah wait, it just doesn't work properly on recursive types:

```
> limitedHash 1 [1..200_001];;
Stack overflow.
Repeat 23958 times:
```

**Comment by cartermp** [comment]
Yep, that's the one.

**Comment by abelbraaksma** [comment]
Do these kind of lib changes to F#Core require an RFC? Or maybe just have an encompassing RFC for all lib changes in version X? (asking since I noticed some versions of 6.0.x introduced new functions not in RFCs, and 7 deprecated some stuff also not in an RFC, which makes it a little hard to keep track).

**Comment by T-Gro** [comment]
@abelbraaksma  : Yes, that is the one. I and I agree with the conclusions. It also matches my finding in the codebase about the functionality not being deeply tested.

I propose we should wait with the actual change for a major version bump.
Then there could be a unifying doc on all the removals, if there are more to be done.

(=> now would be the right time to start collecting ideas for them)

**Comment by edgarfgp** [comment]
Will this be done in the next major version 9 ?
